boa/src/builtins/object/mod.rs

```diff
@@ -289,19 +289,23 @@ impl Object {
 
     /// Define a property in an object
     pub fn define_property(_: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
-        let obj = args.get(0).expect("Cannot get object");
-        let prop = args
-            .get(1)
-            .expect("Cannot get object")
-            .to_property_key(context)?;
-
-        let desc = if let Value::Object(ref object) = args.get(2).cloned().unwrap_or_default() {
-            object.to_property_descriptor(context)?
+        let object = args.get(0).cloned().unwrap_or_else(Value::undefined);
+        if let Some(mut object) = object.as_object() {
+            let key = args
+                .get(1)
+                .unwrap_or(&Value::undefined())
+                .to_property_key(context)?;
+            let desc = args
+                .get(2)
+                .unwrap_or(&Value::undefined())
+                .to_property_descriptor(context)?;
+
+            object.define_property_or_throw(key, desc, context)?;
+
+            Ok(Value::undefined())
```

Comment:
You could also fix this one: `Object.defineProperty` should return the object itself, and not `undefined` (see [spec][spec])

[spec]: https://tc39.es/ecma262/#sec-object.defineproperty

Reply:
> You could also fix this one: `Object.defineProperty` should return the object itself, and not `undefined` (see [spec](https://tc39.es/ecma262/#sec-object.defineproperty))

Nice catch. You are right, I'll fix it now :)